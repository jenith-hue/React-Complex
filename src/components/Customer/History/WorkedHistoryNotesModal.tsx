import {
  makeStyles,
  RACButton,
  RACCOLOR,
  RACModal,
  Typography,
} from '@rentacenter/racstrap';
import React from 'react';

export interface ModalProps {
  workedResultText: string;
  noteText: string;
  open: boolean;
  onClose: () => void;
}
export const useStyles = makeStyles((theme: any) => ({
  dialogContent: {
    textAlign: 'left',
  },
  dialogRoot: {
    '& .MuiDialogContent-root': {
      padding: '1rem',
      flexDirection: 'column',
      display: 'flex',
      height: 'unset',
      gap: theme.typography.pxToRem(20),
    },
    '& .MuiDialog-paperWidthSm': {
      maxWidth: theme.typography.pxToRem(500),
      maxHeight: theme.typography.pxToRem(363),
    },
    '& .MuiTypography-h4': {
      fontSize: '20px !important',
    },
    '& .MuiOutlinedInput-multiline': {
      padding: theme.typography.pxToRem(10),
    },
    '& .MuiInputBase-multiline': {
      margin: '8px 0px 0px 0px !important',
    },
    '& .MuiInputBase-input': {
      fontFamily: 'OpenSans-regular',
      fontSize: theme.typography.pxToRem(14),
    },
  },
  dialogActions: {
    paddingRight: theme.typography.pxToRem(15),
    paddingBottom: theme.typography.pxToRem(15),
    justifyContent: 'center',
  },
  text: {
    fontSize: '1rem !important',
  },
  // same as for text templates -> generated text
  generatedTextValue: {
    borderRadius: '5px',
    minHeight: 'calc(1.5em + 0.75rem + 2px)',
    height: theme.typography.pxToRem(73),
    overflowY: 'auto',
    border: '1px solid #C4C4C4',
    padding: '0.375rem 0.75rem',
    marginBottom: '1rem',
    backgroundColor: RACCOLOR.CULTURED,
  },
}));

export interface WorkedHistoryNotesModalContentProps {
  workedResultText: string;
  noteText: string;
}
const WorkedHistoryNotesModalContent = ({
  workedResultText,
  noteText,
}: WorkedHistoryNotesModalContentProps) => {
  const classes = useStyles();
  return (
    <>
      <span>
        <Typography display="inline" variant="body1" className={classes.text}>
          Worked Result:{' '}
        </Typography>
        <Typography display="inline" variant="body2" className={classes.text}>
          {workedResultText}
        </Typography>
      </span>
      <Typography display="inline" variant="body2" className={classes.text}>
        Additional information on Activity Code
      </Typography>
      <span className={classes.generatedTextValue}>{noteText}</span>
    </>
  );
};

export const WorkedHistoryNotesModal = ({
  workedResultText,
  noteText,
  open,
  onClose,
}: ModalProps) => {
  const classes = useStyles();

  return (
    <RACModal
      isOpen={open}
      titleVariant="h4"
      classes={{
        dialogContent: classes.dialogContent,
        dialog: classes.dialogRoot,
        dialogActions: classes.dialogActions,
      }}
      maxWidth="sm"
      title=""
      content={
        <WorkedHistoryNotesModalContent
          workedResultText={workedResultText}
          noteText={noteText}
        />
      }
      onClose={onClose}
      buttons={
        <>
          <RACButton variant="contained" color="primary" onClick={onClose}>
            OK
          </RACButton>
        </>
      }
    />
  );
};
