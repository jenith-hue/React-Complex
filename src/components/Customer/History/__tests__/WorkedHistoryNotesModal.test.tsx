/* eslint-disable react/display-name */
/* eslint-disable sonarjs/no-identical-functions */
import React from 'react';
import renderer from 'react-test-renderer';

import { WorkedHistoryNotesModal } from '../WorkedHistoryNotesModal';

interface PropWithChildren {
  children: JSX.Element;
}
jest.mock('@rentacenter/racstrap', () => ({
  ...jest.requireActual('@rentacenter/racstrap'),
  Typography: ({ children, ...rest }: PropWithChildren) => (
    <div data-testid="Typography" {...rest}>
      {children}
    </div>
  ),
  RACButton: ({ children, ...rest }: PropWithChildren) => (
    <div data-testid="RACButton" {...rest}>
      {children}
    </div>
  ),
  RACModal: ({ children, ...rest }: PropWithChildren) => (
    <div data-testid="RACModal" {...rest}>
      {children}
    </div>
  ),
  RACCOLOR: {
    CULTURED: 'CULTURED',
  },
  makeStyles: () => () => ({
    dialogContent: 'dialogContent',
    dialogRoot: 'dialogRoot',
    dialogActions: 'dialogActions',
    text: 'text',
    generatedTextValue: 'generatedTextValue',
  }),
}));
describe('WorkedHistoryNotesModal', () => {
  it('should render correctly', async () => {
    const tree = renderer.create(
      <WorkedHistoryNotesModal
        workedResultText="WRTXT"
        noteText="NTXT"
        open
        onClose={jest.fn()}
      />
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
