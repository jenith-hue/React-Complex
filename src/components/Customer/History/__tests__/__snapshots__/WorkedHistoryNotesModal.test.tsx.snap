// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkedHistoryNotesModal should render correctly 1`] = `
<div
  buttons={
    <React.Fragment>
      <RACButton
        color="primary"
        onClick={[MockFunction]}
        variant="contained"
      >
        OK
      </RACButton>
    </React.Fragment>
  }
  classes={
    Object {
      "dialog": "dialogRoot",
      "dialogActions": "dialogActions",
      "dialogContent": "dialogContent",
    }
  }
  content={
    <WorkedHistoryNotesModalContent
      noteText="NTXT"
      workedResultText="WRTXT"
    />
  }
  data-testid="RACModal"
  isOpen={true}
  maxWidth="sm"
  onClose={[MockFunction]}
  title=""
  titleVariant="h4"
/>
`;
